### DISCUSSION TITLE
does flopy.export.utils.export_array_contours() work for disu?

### DISCUSSION THREAD

**Discussion by RyanConway91** [modflowpy/flopy#1953] [Q&A] [answered]
Im working with a octree disu. So though the model is disu, the concept of layers still applies. I get an error when I try and export head contours with flopy.export.utils.export_array_contours. I get why this would not work for an unlayered disu, but I think it should work for an octree grid. 

Is this method not supposed to work for disu or is something else going wrong?

**Comment by wpbonelli** [comment, marked as answer]
I think array contours should work for unstructured grids whether or not layers are the same. Do you mind sharing steps to reproduce?

#1975 adds a couple tests which seem to work. There was not previously a way to select which layer to export, and layer 0 was selected by default. The PR adds support for layer selection.

**Same layers**

![disu_contours](https://github.com/modflowpy/flopy/assets/10502868/3be385bc-dfbe-4b6d-9bfc-947aabf3720d)

**Different layers**

![disu_contours_diff_layers](https://github.com/modflowpy/flopy/assets/10502868/c4381507-a761-43a6-beae-7192560576d4)

**Reply by RyanConway91** [reply]
The snippet I am trying is:

```
a_hd= a_hd_all[-1,:]#get last kstpkper
for l in range(mg.nlay):
    nr = mg.get_layer_node_range(l)    
    a = a_hd[nr[0]:nr[1]]
    flopy.export.utils.export_array_contours(mg,r'%s\hd_cont_L%i.shp'%(d_gis,l+1),a)
```
Looking at the merge you referenced (https://github.com/modflowpy/flopy/pull/1975), it looks like for disu you give it the whole grid array and provide a "layer = layer" arg? (line 726):
 
`export_array_contours(grid, fname, a, layer=layer)
`
When I try this I get "unhandled keywords: {'layer': 0}". I have version '3.5.0.dev0'. Do I just need to update?

Thanks

**Reply by wpbonelli** [reply]
> it looks like for disu you give it the whole grid array and provide a "layer = layer" arg?

Yes, it slices the array to select the layer

It's not merged yet, to try it out you can install the PR branch, sorry about that

**Reply by wpbonelli** [reply]
@RyanConway91 marking this answered but let us know if there are still unresolved issues.
